source/dub/commandline.d

```diff
@@ -97,6 +97,16 @@ int runDubCommandLine(string[] args)
 		environment["TEMP"] = environment["TEMP"].replace("/", "\\");
 	}
 
+	// special stdin syntax
+	if (args.length >= 2 && args[1] == "-")
+	{
+		import dub.internal.utils: getTempFile;
+
+		auto path = getTempFile("app", ".d");
+		std.file.write(path.toNativeString(), stdin.byLineCopy.join);
```

Comment:
FYI if you don't want the the FQSN (std.file), there exists `toFile` just for this reason:

`stdin.byChunk(4096).joiner.toFile(path.toNativeString())`

Also use `byChunk` instead of `byLineCopy` to avoid unnecessary allocation + copying and as a bonus you don't loose the line endings.
Furthermore with `joiner` you avoid allocations :)

Reply:
great! looks much better now.

Reply:
DMD 2.071.2 and lower does not support toFile ):